modules/gh-oidc/README.md

```diff
@@ -0,0 +1,111 @@
+
+- Creation of a Workload Identity pool
+- Configuring a Workload Identity provider
+- Granting external identities necessary IAM roles on Service Accounts
+
+### Example Usage
+
+```terraform
+module "gh_oidc" {
+  source      = "terraform-google-modules/github-actions-runners/google//modules/gh-oidc"
+  project_id  = var.project_id
+  pool_id     = "example-pool"
+  provider_id = "example-gh-provider"
+  sa_mapping = {
+    "foo-service-account" = {
+      sa_name   = "foo-service-account@my-project.iam.gserviceaccount.com"
+      attribute = "attribute.repository/user/repo"
+    }
+  }
+}
+```
+
+Below are some examples:
+
+### [OIDC Simple](../../examples/oidc-simple/README.md)
+
+This example shows how to use this module along with a Service Account to access storage buckets.
+
+### GitHub Workflow
+
+Once provisioned, you can use the [oidc-auth-google-cloud](https://github.com/sethvargo/oidc-auth-google-cloud) Action in a workflow as shown below
+
+```yaml
+# Example workflow
+# .github/workflows/example.yml
+
+name: 'example oidc'
+on:
+  push:
+    branches:
+    - 'main'
+jobs:
+  run:
+    name: 'example to list bucket contents'
+    permissions:
+      id-token: write
+      contents: read
+    runs-on: 'ubuntu-latest'
+    steps:
+    - id: 'auth'
+      uses: 'sethvargo/oidc-auth-google-cloud@main'
+      with:
+        workload_identity_provider: ${{ secrets.PROVIDER_NAME }} # this is the output provider_name from the TF module
+        service_account: ${{ secrets.SA_EMAIL }} # this is a SA email configured using the TF module with access to YOUR-GCS-BUCKET
+    - id: 'list buckets contents'
+      run: |-
+        curl https://storage.googleapis.com/storage/v1/b/YOUR-GCS-BUCKET/o \
+          --header "Authorization: Bearer ${{ steps.auth.outputs.access_token }}"
+```
+
+<!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
+## Inputs
+
+| Name | Description | Type | Default | Required |
+|------|-------------|------|---------|:--------:|
+| allowed\_audiences | Workload Identity Pool Provider allowed audiences. Currently, GitHub only allows sigstore | `list(string)` | <pre>[<br>  "sigstore"<br>]</pre> | no |
+| attribute\_condition | Workload Identity Pool Provider attribute condition expression. [More info](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/iam_workload_identity_pool_provider#attribute_condition) | `string` | `null` | no |
+| attribute\_mapping | Workload Identity Pool Provider attribute mapping. [More info](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/iam_workload_identity_pool_provider#attribute_mapping) | `map(any)` | <pre>{<br>  "attribute.actor": "assertion.actor",<br>  "attribute.aud": "assertion.aud",<br>  "attribute.repository": "assertion.repository",<br>  "google.subject": "assertion.sub"<br>}</pre> | no |
+| issuer\_uri | Workload Identity Pool Provider issuer uri. | `string` | `"https://vstoken.actions.githubusercontent.com"` | no |
```

Comment:
What's the use case for making this configurable?

Reply:
Hardcoded this for now, I am not sure if this hold true for GHES and hosted runners but we can expose in the future if needed.